### PR TITLE
(0.89.0) Better `Simulation` interface and some miscellaneous improvements + remove `MEWSVerticalDiffusivity`

### PR DESCRIPTION
**Pull request by glwagner**
This PR tweaks the interface for `Simulation` to make it easier to build new models. These improvements will help [`ClimaSeaIce` ](https://github.com/CliMA/ClimaSeaIce.jl) implement a new sea ice model that can leverage `Oceananigans.Simulation`.

I've also included a few minor miscellaneous improvements to kernel launching + documentation.

This PR deletes MEWSVerticalDiffusivity (which nobody used and is not planned to be developed anymore).

### CODE REVIEW

File: src/Utils/kernel_launching.jl [hunk 1]
```diff
@@ -6,15 +6,31 @@ using Oceananigans.Architectures
...
+size = (8, 6, 4)
+offsets = (0, 1, 2)
+kp = KernelParameters(size, offsets)
+
+# Launch a kernel with indices that range from i=1:8, j=2:7, k=3:6,
+# where i, j, k are the first, second, and third index, respectively:
+launch!(arch, grid, kp, kernel!, kernel_args...)
+```
+"""
 KernelParameters(size, offsets) = KernelParameters{size, offsets}()
```
Comment (glwagner) [inline src/Utils/kernel_launching.jl:31]: @simone-silvestri please advice if this docstring makes sense

Comment (simone-silvestri) [reply src/Utils/kernel_launching.jl:35]: makes sense

File: src/Utils/kernel_launching.jl [hunk 2]
```diff
@@ -94,10 +117,10 @@ function launch!(arch, grid, workspec, kernel!, kernel_args...;
...
                  kwargs...)
 
-    workgroup, worksize = work_layout(grid, worktuple(workspec);
-                                      include_right_boundaries,
-                                      reduced_dimensions,
-                                      location)
+    workgroup, worksize = work_layout(grid, workspec;
+                                     include_right_boundaries,
+                                     reduced_dimensions,
+                                     location)
```
Comment (navidcy) [inline src/Utils/kernel_launching.jl:123]:
```suggestion
                                      include_right_boundaries,
                                      reduced_dimensions,
                                      location)
```

seems that the alignment went bit off here?

File: src/Utils/kernel_launching.jl [hunk 3]
```diff
@@ -6,15 +6,35 @@ using Oceananigans.Architectures
...
+size = (8, 6, 4)
+offsets = (0, 1, 2)
+kp = KernelParameters(size, offsets)
+
+# Launch a kernel with indices that range from i=1:8, j=2:7, k=3:6,
+# where i, j, k are the first, second, and third index, respectively:
+launch!(arch, grid, kp, kernel!; kernel_args...)
+```
+
+See the documentation for `launch!`.
```
Comment (navidcy) [inline src/Utils/kernel_launching.jl:33]:
```suggestion
See the documentation for [`launch!`](@ref Oceananigans.Utils.launch!).
```

**Comment by navidcy** [comment]
@glwagner is this a breaking change?

**Comment by navidcy** [comment]
> @glwagner is this a breaking change?

I guess since MEWS went away... yes

**Comment by navidcy** [comment]
@glwagner some `simulation-tests` are failing. Could you address that?

**Review by navidcy — APPROVED** [review]

**Comment by tomchor** [comment]
@glwagner is this PR breaking only because of the deletion of `MEWSVerticalDiffusivity`?

Or is there something else that's breaking in the `Simulation` interface?

**Comment by glwagner** [comment]
Technically, if you define a custom `AbstractModel` then this does represent a breaking change. But it's not breaking for setups that use only Oceananigans models.

**Pull request merged**